Makefile.targets.mk

```diff
@@ -12,24 +59,39 @@ REV=$(shell git rev-parse --short HEAD)
 # Tag is the tag to use for build and push image targets.
 TAG ?= $(REV)
 
-.PHONY: build
-build:  ## Build the envoy-gateway binary
-	@CGO_ENABLED=0 go build -a -o ./bin/${GOOS}/${GOARCH}/ github.com/envoyproxy/gateway/cmd/envoy-gateway
+# Build the envoy-gateway binaries in target platforms
+.PHONY: build.%
+build.%:
+	$(eval COMMAND := $(word 2,$(subst ., ,$*)))
```

Comment:
this magic reduces readability, can you clarify what the benefits are ?

Reply:
1. build.% is dynamic, we do not need to hardcode platform and cmds like build-linux-amd64 or build-linux-arm64 and even build-darwin-amd64 build-darwin-arm64 for envoy-gateway, we just need to excute `make build.{Platform}.{CMD}` like `make build.linux_amd64.envoy-gateway`
2. if we add another cmd except `envoy-gateway` in `/cmd/**` like `envoy-gateway2`, we do not need write a couple make file commands for it, just execute `make build.linux_amd64.envoy-gateway2`

Reply:
I think some comments outlining what you're saying would make this much easier to understand. Perhaps a comment here saying that this is a dynamic build target, and that you should use `make build.linux_amd64.envoy-gateway` or something. And we should have something such that `make help` will give you a list of options for `make build.whatever`.

Some comments on the platform-selection logic above would also make this clearer. I personally haven't seen this before, but it's a neat trick, that we definitely need to explain.